pkg/controller/registry/reconciler/configmap.go

```diff
@@ -513,5 +515,39 @@ func (c *ConfigMapRegistryReconciler) CheckRegistryServer(logger *logrus.Entry,
 	}
 
 	healthy = true
+	if deadPodsDetected, livePodsCount, e := detectAndDeleteDeadPods(logger, c.OpClient, pods, source.GetNamespace()); deadPodsDetected {
+		if livePodsCount == 0 {
+			healthy = false
+		}
+		err = e
+	}
```

Comment:
Nit: do we really care if dead pods are detected? Seems like this could be simplified to:
```go
livePodsCount, deleteDeadPodsErr := detectAndDeleteDeadPods(logger, c.OpClient, pods, source.GetNamespace())
if deleteDeadPodsErr != nil {
	return false, deleteDeadPodsErr
}
return livePodsCount > 0, nil
```

Reply:
In fact, we don't even care how many live pods there are. We could have `detectAndDeleteDeadPods` return `(bool, error)`, where the boolean value is `haveAnyAlivePods`.

Reply:
Fair.